src/groth16.cpp

```diff
@@ -291,10 +291,14 @@ Prover<Engine>::prove(typename Engine::FrElement* wtns)
     E.fr.copy(r, E.fr.zero());
     E.fr.copy(s, E.fr.zero());
 
-    // randombytes_buf((void*)&(r.v[0]), sizeof(r) - 1); // TODO(zi): WHY -1
-
-    fill_with_random_bytes(r);
-    fill_with_random_bytes(s);
+    // Filling in the last byte here with a non-zero value causes a small amount of proofs to fail,
+    // possibly due to overflowing the field modulus
+    randombytes_buf((void*)&(r.v[0]), sizeof(r) - 1);
+    randombytes_buf((void*)&(s.v[0]), sizeof(s) - 1);
+
+    // Make extra sure the final byte is 0
+    reinterpret_cast<char*>(&r)[sizeof(r) - 1] = 0;
+    reinterpret_cast<char*>(&s)[sizeof(s) - 1] = 0;
```

Comment:
are `r` and `s` supposed to be uniformly distributed in [0,q-1]?
does this only guarantee uniform distribution in `[0,2^k-1]`?